lib/ruby_lsp/tapioca/lockfile_diff_parser.rb

```diff
@@ -0,0 +1,44 @@
+# typed: true
+# frozen_string_literal: true
+
+module RubyLsp
+  module Tapioca
+    class LockfileDiffParser
+      GEM_PATTERN = /[+-](.*[\w\-]+)\s*\(/
+      ADDED_LINE_PATTERN = /^\+.*[\w\-]+ \(.*\)/
+      REMOVED_LINE_PATTERN = /^-.*[\w\-]+ \(.*\)/
+
+      attr_reader :added_or_modified_gems
+      attr_reader :removed_gems
+
+      def initialize(diff_content)
+        @diff_content = diff_content
```

Comment:
```suggestion
        @diff_content = diff_content.lines
```

If the only thing we do is call `.lines` on it we could do it here and do it only once instead of 2 calls below.